src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -694,6 +694,16 @@ ReplicatedRecoveryBackend::_handle_pull_response(
         recovery_waiter.obc = obc;
         obc->obs.oi.decode_no_oid(push_op.attrset.at(OI_ATTR), push_op.soid);
         pull_info.recovery_info.oi = obc->obs.oi;
+
+        auto ss_attr_iter = push_op.attrset.find(SS_ATTR);
+        if (ss_attr_iter != push_op.attrset.end()) {
+          if (!obc->ssc) {
+            obc->ssc = new crimson::osd::SnapSetContext(
+              push_op.soid.get_snapdir());
+          }
+          obc->ssc->snapset = SnapSet(ss_attr_iter->second);
```

Comment:
If we decide not to edit `ObjectContextLoader::with_obc` (previous comment) which eventually calls `PGBackend::load_metadata` to initiate ssc - the SnapSet constructor should be in try-catch block since it's being decoded.

Reply:
It seems that `PGBackend::load_metadata` has already made sure that SnapSet ctor wouldn't throw exceptions, am I right? https://github.com/ceph/ceph/blob/b05d167b48ec517e1006dee9543995d0c8a1bda2/src/crimson/osd/pg_backend.cc#L108C7-L126

Reply:
Yes. Although I was referring to the additional call (`SnapSet(ss_attr_iter->second);`) which is not part of `PGBackend::load_metadata`.

Reply:
> Yes. Although I was referring to the additional call (`SnapSet(ss_attr_iter->second);`) which is not part of `PGBackend::load_metadata`.

Um, the call `SnapSet(ss_attr_iter->second);` is also enclosed by the condition `ss_attr_iter != push_op.attrset.end()`, so it also won't throw exceptions, am I right?

Reply:
```
void SnapSet::decode(ceph::buffer::list::const_iterator& bl)
{
  DECODE_START_LEGACY_COMPAT_LEN(3, 2, 2, bl);
  decode(seq, bl);
  bl += 1u;  // skip legacy head_exists (always true)
  decode(snaps, bl);
  decode(clones, bl);
  decode(clone_overlap, bl);
  decode(clone_size, bl);
  if (struct_v >= 3) {
    decode(clone_snaps, bl);
  } else {
    clone_snaps.clear();
  }
  DECODE_FINISH(bl);
}
```

I think that the decoding can throw exceptions if the buffer got corrupted somehow.

Reply:
Yes, in this case, I think we should catch the exception outside SnapSet's ctor instead of in it, what do you think?

Reply:
I had similar suggestion in the past. @athanatos, what is your opinion?

Reply:
Probably easiest to just add a try/catch here?

Reply:
Done, please take a look again.